File: numeroarstromng/narcicista.java
package numeroarstromng;
import javax.swing.JOptionPane;


public class narcicista {
    public static void main(String[] args) {
        
    String n = JOptionPane.showInputDialog(null,"Ingresa un numero:");
    int [] num = new int[n.length()];    
    int r = 0;

    for (int i = 0; i < n.length(); i++) {
        num[i] = Character.getNumericValue(n.charAt(i));
        r+=Math.pow(num[i],n.length());
    }

    if(r==Integer.parseInt(n)){

        System.out.println("Es un numero narcicista");

    }else{

        System.out.println("No es un numero narcicista");

    }

}

}

File: numeroarstromng/narcicista.py

n = input("Ingresa un numero: ")
r = 0 

for i in range(0,len(n)):
    r+=int(n[i])**len(n) 

if(r==int(n)):
    print("Es un numero narcicista");
else:
    print("No es un numero narcicista");